<launch>
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <arg name="color_mesh_by_label" default="false" doc="display mesh colored by semantic label"/>

    <arg name="robot_frame" default="base_link" doc="robot body tf frame"/>
    <arg name="odom_frame" default="odom" doc="odometry (map) frame"/>
    <arg name="map_frame" default="odom"/>
    <arg name="sensor_frame" default="left_cam" doc="camera frame"/>
    <arg name="hydra_config_path" default="$(find hydra)/config/datasets/oakd.yaml"/>
    <arg name="labelspace_name" default="ade20k_mit" doc="semantic label space"/>

    <group>
        <remap from="semantic_inference/color/image_raw/compressed" to="/oak/rgb/image_raw/compressed"/>
        <include file="$(find semantic_inference_ros)/launch/semantic_inference.launch" pass_all_args="true">
            <arg name="rgb_image_transport" value="compressed"/>
        </include>
    </group>

    <arg name="start_hydra" default="true"/>
    <group if="$(arg start_hydra)">
        <remap from="hydra_ros_node/input/camera/depth_registered/image_rect" to="/oak/stereo/image_raw"/>
        <remap from="hydra_ros_node/input/camera/rgb/image_raw/compressed" to="/oak/rgb/image_raw/compressed"/>
        <remap from="hydra_ros_node/input/camera/rgb/camera_info" to="/oak/rgb/camera_info"/>
        <remap from="hydra_ros_node/input/camera/semantic/image_raw" to="semantic_inference/semantic/image_raw"/>
        <param name="hydra_ros_node/input/camera/rgb/image_transport" value="compressed"/>
        <include file="$(find hydra_ros)/launch/hydra.launch" pass_all_args="true">
            <arg name="dataset_name" default="oakd"/>
        </include>
    </group>

    <!-- visualizer for converting dsg to visualization markers -->
    <arg name="start_visualizer" default="true" doc="start hydra visualizer"/>
    <group if="$(arg start_visualizer)">
        <remap from="hydra_dsg_visualizer/dsg" to="hydra_ros_node/backend/dsg"/>
        <remap from="hydra_dsg_visualizer/feature" to="hydra_ros_node/input/left_cam/feature"/>
        <include file="$(find hydra_visualizer)/launch/hydra_streaming_visualizer.launch" pass_all_args="true">
            <arg name="visualizer_frame" default="odom"/>
        </include>
    </group>

    <!-- rviz -->
    <arg name="start_rviz" default="true" doc="start rviz"/>
    <group if="$(arg start_rviz)">
        <include file="$(find hydra_visualizer)/launch/hydra_rviz.launch">
            <arg name="rviz_dir" default="$(find hydra_ros)/rviz"/>
            <arg name="rviz_file" default="oakd.rviz"/>
        </include>
    </group>

    <!-- kimera -->
    <arg name="start_kimera" default="true" doc="start kimera"/>
    <group if="$(arg start_kimera)">
        <include file="$(find hydra_vio_configs)/launch/oakd_rgbd_vio.launch">
            <arg name="start_rviz" value="false"/>
        </include>
    </group>

</launch>
